olp-cpp-sdk-dataservice-write/test/unit/TestStreamLayerClient.cpp

```diff
@@ -1923,44 +1972,48 @@ TEST_P(StreamLayerClientCacheMockTest, FlushDataMultiple) {
   for (auto& single_response : response) {
     ASSERT_NO_FATAL_FAILURE(PublishDataSuccessAssertions(single_response));
   }
+  testing::Mock::VerifyAndClearExpectations(network_.get());
 }
 
-#if 0
-TEST_P(StreamLayerClientCacheMockTest, FlushDataCancel) {
-  auto cancel_token = olp::client::CancellationToken();
-  ON_CALL(handler_,
-          CallOperator(IsGetRequest(URL_LOOKUP_CONFIG), testing::_,
-          testing::_))
-      .WillByDefault(testing::DoAll(
-          testing::InvokeWithoutArgs(
-              [&cancel_token]() { cancel_token.cancel(); }),
-          testing::Invoke(
-              returnsResponse({200, HTTP_RESPONSE_LOOKUP_CONFIG}))));
+TEST_P(StreamLayerClientCacheMockTest, DISABLED_FlushDataCancel) {
+  auto waitForCancel = std::make_shared<std::promise<void>>();
+  auto pauseForCancel = std::make_shared<std::promise<void>>();
+
+  olp::http::RequestId request_id;
+  NetworkCallback send_mock;
+  CancelCallback cancel_mock;
+
+  std::tie(request_id, send_mock, cancel_mock) = generateNetworkMocks(
+      waitForCancel, pauseForCancel, {200, HTTP_RESPONSE_LOOKUP_CONFIG});
 
   {
     testing::InSequence dummy;
 
-    EXPECT_CALL(handler_, CallOperator(IsGetRequest(URL_LOOKUP_INGEST),
-                                       testing::_, testing::_))
-        .Times(1);
-    EXPECT_CALL(handler_, CallOperator(IsGetRequest(URL_LOOKUP_CONFIG),
-                                       testing::_, testing::_))
+    EXPECT_CALL(*network_, Send(IsGetRequest(URL_LOOKUP_INGEST), _, _, _, _))
         .Times(1);
+    EXPECT_CALL(*network_, Send(IsGetRequest(URL_LOOKUP_CONFIG), _, _, _, _))
+        .Times(1)
+        .WillOnce(testing::Invoke(std::move(send_mock)));
+    EXPECT_CALL(*network_, Cancel(request_id))
+        .WillOnce(testing::Invoke(std::move(cancel_mock)));
   }
 
   auto error = client_->Queue(
       PublishDataRequest().WithData(data_).WithLayerId(GetTestLayer()));
 
   ASSERT_FALSE(error) << error.get();
 
-  auto cancel_future = client_->Flush();
-  cancel_token = cancel_future.GetCancellationToken();
-  auto response = cancel_future.GetFuture().get();
+  auto promise = client_->Flush();
+  waitForCancel->get_future().get();
+  promise.GetCancellationToken().cancel();
+  pauseForCancel->set_value();
+
+  auto response = promise.GetFuture().get();
 
   ASSERT_EQ(1, response.size());
+  testing::Mock::VerifyAndClearExpectations(network_.get());
```

Comment:
Adding additionally this is not really necessary as google does it for you https://github.com/google/googletest/blob/master/googlemock/docs/cheat_sheet.md#verifying-and-resetting-a-mock
-> *gMock will verify the expectations on a mock object when it is destructed ...*

Reply:
sometimes network mock might be never destroyed as reference to is is hold by expectation inside the mock

Reply:
Then move it to TearDown() please

Reply:
In my opinion TearDown is not the place to verify assertions or expectations on mocks event.

Reply:
En contraire mon frere, see https://github.com/google/googletest/blob/master/googletest/docs/faq.md#should-i-use-the-constructordestructor-of-the-test-fixture-or-setupteardown-ctorvssetup

Reply:
I didn't get that. It tells that I might want to prefer TearDown over destructors. Nothing there tells that VerifyAndClearExpectations should be preferably called in TearDown and there are plenty of examples when it called in tests. Placing this call inside TearDown might help code deduplication, but believe having all assertions in one place at the buttom of each test is more readable and less astonishing.